### PR TITLE
Add a test to ensure that calling Set after Close does not panic

### PR DESCRIPTION
**Pull request by josephschorr**
On a previous version of ristretto, we experienced a panic occasionally due to calling Set after Close resulting in a write to the closed `setBuf` channel. We originally wrote this test as part of a fix on a forked version of Ristretto. Thankfully, on HEAD, this test is no longer failing, so we are submitting it to act as a guard against any regressions in the future.

NOTE: The update to `golang.org/x/sys` also fixes a build issue with Golang 1.18 on Darwin

---
This change is [Reviewable](https://reviewable.io/reviews/outcaste-io/ristretto/5)

### CODE REVIEW

**Comment by jzelinskie** [comment]
@manishrjain @jarifibrahim any updates? We'd like to migrate off our fork to the new upstream.

**Review by jarifibrahim — APPROVED** [review]
hi @josephschorr @jzelinskie 👋 ! Thanks for the PR. 
The change looks good to me. I cannot merge this PR because I no longer have write access to this repository. 
@akon-dey can review and merge this PR.

**Pull request merged**
